docs/python-examples/fixtures/test_fixtures.py

```diff
@@ -0,0 +1,109 @@
+import logging
+
+import pytest
+
+logger = logging.getLogger(__name__)
+
+
+def alter_pvc(pvc):
+    """
+    Dummy helper function to alter pvc
+
+    Args:
+        pvc (PVC): PVC object reference
+    """
+    logger.info(f"Doing something with pvc: {pvc}")
+
+
+@pytest.fixture(scope='class')
+def three_pvcs(pvcs_factory, storage_class):
+    """
+    This is fixture related just to one specific test in this module so no
+    need to have it defined in conftest
+    """
+    return pvcs_factory(3, storage_class)
+
+
+class TestCreatingPVCsFromTest:
+    # This needs to be discussed! How to solve the issue of sharing resources
+    # created by factory fixtures in class level scope.
+    shared_pvcs = []
+
+    @pytest.mark.parametrize("pvcs_number", (2, 4))
+    def test_create_pvcs(self, pvcs_number, pvcs_factory, storage_class):
+        """
+        You can access all needed resources via fixtures as parameters in
+        method definition like in example above (pvcs_factory or
+        storage_class).
+        """
+        pvcs_created = pvcs_factory(pvcs_number, storage_class)
+        len(pvcs_created) == pvcs_number
+        logger.info([p.name for p in pvcs_created])
+
+    def test_share_pvcs(self, pvcs_factory, storage_class):
+        my_shared_pvcs = pvcs_factory(2, storage_class)
+        logger.info(f"Shared pvcs usage: {[p.name for p in my_shared_pvcs]}")
+
+        # Is this acceptable to do below for share PVCs to test 3rd test?
```

Comment:
I think if multiple tests need to 'share' the output from a class scope factory fixture, we should create a wrapper fixture that uses the factory to create the shared resources and returns the list of them. That way the tests that need the list of resources can just use the wrapper fixture to get the list. For example these tests would use the`three_pvcs` fixture to get the list of 3 pvcs and not the `pvc_factory` fixture directly.

Reply:
What about case those resources needs to be created as a part of test?

Reply:
In this scenario you have one test that needs to create the resources, and other tests that need to utilize those created in the other test. If that's the case then sure using a class attribute to pass those created resources from one test to another is fine. As long as the fixture isn't altering that class attribute directly and it's done at the test level like you have here.

Reply:
If a fixture is a ``class`` scope you still won't need that class attribute though. If the first test uses the ``class`` level fixture and modifies it, then the next class will get that modified version. But to @clacroix12 point, using a class attribute is still a feasible solution if done inside a test explicitly. That's a lot less confusing than fixtures that magically change class attributes.

Reply:
@andrewschoen problem here is that the fixture factory returns the function, so when you want to create some resources by this function in one of the test and in the other test you simply cannot get the same resources as if you will use create_pvcs_factory then you will get the same instance of function for the class but not the resources created in previous call from different test in that same class.